query-engine/core/src/schema_builder/input_type_builder/update_input_type_extension.rs

```diff
@@ -365,4 +369,10 @@ pub trait UpdateInputTypeBuilderExtension<'a>: InputTypeBuilderBase<'a> + Create
         input_object.set_fields(fields);
         Arc::downgrade(&input_object)
     }
+
+    fn field_should_be_kept_for_update_input_type(field: &ScalarFieldRef) -> bool {
+        // We forbid updating auto-increment integer ID fields as this can create problems with the
+        // underlying sequences. We do the same for creates.
+        !field.is_auto_generated_int_id
+    }
```

Comment:
Question: I don't have a great intuition for that part of the codebase yet, This is mostly repeated from [the CreateInputTypeExtension trait](https://github.com/prisma/prisma-engines/blob/225c75492b90e646e0224803ae30926eb676bdd0/query-engine/core/src/schema_builder/input_type_builder/create_input_type_extension.rs#L166). Do you think it would make sense to try and avoid repetition here?

Reply:
Keep it like you implemented it for now.

Reply:
I am personally ok with the repetition. I don't have a strong opinion about this.